cmd/apiserver/app/server/run_server_test.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+import (
+	"errors"
+	"testing"
+
+	kubeclientfake "k8s.io/kubernetes/pkg/client/clientset_generated/release_1_5/fake"
+	"k8s.io/kubernetes/pkg/client/testing/core"
+	genericserveroptions "k8s.io/kubernetes/pkg/genericapiserver/options"
+	"k8s.io/kubernetes/pkg/runtime"
+)
+
+// make sure RunServer returns with an error when TPR fails to install
+func TestRunServerInstallTPRFails(t *testing.T) {
+	options := &ServiceCatalogServerOptions{
+		GenericServerRunOptions: &genericserveroptions.ServerRunOptions{},
+		SecureServingOptions: &genericserveroptions.SecureServingOptions{
+			ServingOptions: genericserveroptions.ServingOptions{
+				BindPort: 8888,
+			},
+		},
+	}
+
+	fakeClientset := &kubeclientfake.Clientset{}
+	fakeClientset.AddReactor("create", "thirdpartyresources", func(core.Action) (bool, runtime.Object, error) {
```

Comment:
@pmorie is there a reactor call to make sure that this has actually been called? I don't see anything like this in the top-level faking code, but there's so much code I might just be overlooking it.

Since this is a mocking concept, possibly known as 'strict mocking' or 'expectation checking' or 'ensure callers'.